kotlin/PixelPass/src/commonMain/kotlin/io/mosip/pixelpass/PixelPass.kt

```diff
@@ -139,26 +142,40 @@ class PixelPass {
         while (iterator.hasNext()){
             val next = iterator.next()
             val key = mapper[next] ?: next
-            val value = jsonData.get(next)
-            payload.put(key,value)
+            val value = jsonData.get(next.toString())
+            payload.put(key.toString(),value)
         }
         return payload.toString()
     }
-    private fun toBitmap(qrCode: QrCode): Bitmap {
-        Objects.requireNonNull(qrCode)
-        require(!(QR_SCALE <= 0 || QR_BORDER < 0)) { "Value out of range" }
-        require(!(QR_BORDER > Int.MAX_VALUE / 2 || qrCode.size + QR_BORDER * 2L > Int.MAX_VALUE / QR_SCALE)) { "Scale or border too large" }
-        val result = Bitmap.createBitmap(
-            (qrCode.size + QR_BORDER * 2) * QR_SCALE,
-            (qrCode.size + QR_BORDER * 2) * QR_SCALE,
-            Bitmap.Config.ARGB_8888
-        )
-        for (y in 0 until result.getHeight()) {
-            for (x in 0 until result.getWidth()) {
-                val color = qrCode.getModule(x / QR_SCALE - QR_BORDER, y / QR_SCALE - QR_BORDER)
-                result.setPixel(x, y, if (color) Color.BLACK else Color.WHITE)
-            }
+
+
+    fun decodeCWT(
+        cwt: String,
+        publicKeyString: String,
+        mapper: Map<String, String>,
+        philSysPrefix: ArrayList<String>,
+        algorithm: String
+    ): String {
+
+        val isPhilSysData = Util().isPhilSysQRData(cwt, philSysPrefix)
+        val splittedData: String = cwt.substring(cwt.indexOf(":") + 1)
+        val base45DecodedData = Base45.getDecoder().decode(splittedData)
+        val oneKey = KeyUtil.oneKeyFromPublicKey(publicKeyString, algorithm, isPhilSysData)
+        return verifyAndDecode(base45DecodedData, oneKey, mapper)
+    }
+
+    private fun verifyAndDecode(rawCbor: ByteArray, oneKey: OneKey, mapper: Map<String, String>): String {
+        try {
+            val ctx: CwtCryptoCtx = CwtCryptoCtx.sign1Verify(oneKey.PublicKey())
```

Comment:
why in philsys QR code alone we do the verification also here ? Is that not the responsibility of vc-verifier ?